src/component/RegisterForm.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import {Checkinput} from './Checkinput';
+import {UserInfo} from './UserInfo';
+
+class RegistrationForm extends React.Component{
+	timer;
+	constructor(props){
+		super();
+		this.state={
+			data:{
+				name:"",
+				psw:"",
+				confirmpsw:"",
+				phone:"",
+			},	
+			permittedname:false,
+			permittedpsw:false,
+			confirmedpsw:false,
+			permittedphone:false,
+			submitted:false,
+		}
+		// this.timer = "";
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.checkName = this.checkName.bind(this);
+		this.checkPsw = this.checkPsw.bind(this);
+		this.confirmPsw = this.confirmPsw.bind(this);
+		this.checkPhone = this.checkPhone.bind(this);
+	}
+	checkName(e){
+		const namestr = e.target.value;
+		clearTimeout(this.timer);
+		this.timer = setTimeout((namestr) => {
+			const name = this.delHtmlTag( namestr );
+			if(name !== ""){
+				this.setState((prestate)=>{
+					prestate.data.name = name;
+					prestate.permittedname = true;
+					prestate.submitted = false;
+					return prestate;
+				})
+			}else{
+				this.setState((prestate)=>{
+					prestate.data.name = name;
+					prestate.permittedname = false;
+					prestate.submitted = false;
+					return prestate;
+				})
+			}
+		} ,1000 ,namestr);
+		
+	}
+	checkPsw(e){
+		const pswstr = e.target.value;
+		clearTimeout(this.timer);
+		this.timer = setTimeout((pswstr)=>{
+			const psw = this.delHtmlTag( pswstr );
+			const regUpper = /[A-Z]/;
+			const reglower = /[a-z]/;
+			if(regUpper.test(psw) && reglower.test(psw) && psw.length >= 8){
```

Comment:
try to merge if... else...together, and try to don't setState more than once for one action.